src/amo/components/Header/index.js

```diff
@@ -61,14 +62,15 @@ export class HeaderBase extends React.Component {
 
   renderMenuOrAuthButton() {
     const {
+      withBlogUI,
       i18n,
       isReviewer,
       loadedPageIsAnonymous,
       siteIsReadOnly,
       siteUser,
     } = this.props;
 
-    if (loadedPageIsAnonymous) {
+    if (withBlogUI || loadedPageIsAnonymous) {
```

Comment:
Can't we just pass `loadedPageIsAnonymous` in?  (Or rename the parameter to identify what it's actually doing)

Reply:
Or just not call `renderMenuOrAuthButton` at all when `withBlogUI == True` like we do with including `<SearchForm>` below

Reply:
possible yes

Reply:
so I looked at the code again and `loadedPageIsAnonymous` is a mapped prop, we cannot override it, it is derived from the redux state. As for `withBlogUI` and `renderMenuOrAuthButton`, when there is a method that renders something in `render()`, we usually call the method all the time and the method decides if it should render something or early return.

Reply:
mmm, okay